CDCgov/prime-reportstream
# 8752 Added CA receiver transforms to truncate fields needed for CA

This PR adds CA receiver transforms to truncate the HL7 fields needed for CA.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Send the attached FHIR bundle to the Reports API:
[Simple Report.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11165604/Simple.Report.postman_collection.txt)
This sample contains all the fields that should be truncated and you can search for them by looking for "truncatethis"

2. Verify that the request was accepted

3. Open the generated HL7 file in the local sftp folder.

4. Verify all the fields were truncated by making sure the HL7 message doesn't include the text "truncatethis"
 
5. Run integration tests and make sure they pass

## Changes
- Added CA receiver transforms to truncate HL7 fields

## Checklist

### Testing

### Process

## Linked Issues
- #8752 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
